source/hic_hal/stm32/stm32h743ii/extended_features/include/DAP_vendor_ex.h

```diff
@@ -11,6 +11,8 @@ typedef enum
     ID_DAP_VendorEx37_HOLD_IN_BL,
     ID_DAP_VendorEx38_RESET_DAPLINK,
     ID_DAP_VendorEx39_READ_UDC_ADAPTER_TYPE_ADC,
+    ID_DAP_VendorEx40_MEASURE_POWER,
+    ID_DAP_VendorEx41_READ_POWER,
```

Comment:
To provide a snapshot of all measurements when I issue a `measure power` command.
- In pac193x, I need to call `refresh` command to measure the power and then read the voltage and current using a separate command. If only one command is used,  I have to call `refresh` for each request. That is, if I use  two commands `read voltage` and `read current`, they will get the measurement from two different time, which I think is undesirable.

Reply:
Sorry for the late reply. I pressed the wrong button so that github pended my comment.

Reply:
In current I2C implementation, there is a `ARM_I2C_BUS_CLEAR` before each I2C transfer(`hic_hal/stm32/stm32h743ii/STM32H7xx_HAL_Driver/Src/I2C_STM32H7xx.c: I2C_Control`), I see a HAL_Delay(I2C_BUS_CLEAR_CLOCK_PERIOD), which would have 2ms delay.

Reply:
@eleenest, who will use this feature. Does it matter how these functions are called?

Reply:
The intention is to be able to add "simple power measurememts" of DUT's when we can.  
For DUT's that get power from USB-A ports then this could be sampled as part of a test, as well as keeping an eye on the current drawn by the UDB itself.  
For DUT's that are powered from other voltages (12V, 24V ac, etc. some future adaptor would be needed).  

So  I think it is ok to get the initial implementation in, and refine as needed.  


Note: If we are doing a series of I2C transactions right after each other the ARM_I2C_BUS_CLEAR is probably redundant.  
That can help in cases where some I2C peripheral has gotten into a funky state / some UDB crash (where an I2C transaction gets stuck in the middle).  
Maybe we can pass in a flag to determine if we should call **ARM_I2C_BUS_CLEAR**, and normally only send it when starting up UDB (maybe do an initial measurement of the power chip once to set things up / clear the bus).

Reply:
From the datasheet, my understanding is that normally the chip is constantly doing ADC measurements.  When a "refresh" command is sent, then I think it copies the "latest values" to registers, and 1 msec later you can read the registers (they are a snapshot from when the refresh command was sent, I think).  

So potentially we could do both the "refresh" and "reading" in a single command, if we are willing to block for an additional 1 msec?
i.e. if sending the refresh might be 0.5 msec?, then a delay of 1msec, then reading a block of values for an additional 0.5 msec?

Or, we leave it as 2 commands, where NDM first sends a "measure_now", and then a "read_measurement" command later.